src/scripting/lua_kernel_base.cpp

```diff
@@ -501,6 +511,179 @@ static int intf_get_language(lua_State* L)
 	return 1;
 }
 
+static void dir_meta_helper(lua_State* L, std::vector<std::string>& keys)
+{
+	switch(luaL_getmetafield(L, -1, "__dir")) {
+		case LUA_TFUNCTION:
+			lua_pushvalue(L, 1);
+			lua_push(L, keys);
+			lua_call(L, 2, 1);
+			keys = lua_check<std::vector<std::string>>(L, -1);
+			break;
+		case LUA_TTABLE:
+			auto dir_keys = lua_check<std::vector<std::string>>(L, -1);
+			std::copy(dir_keys.begin(), dir_keys.end(), std::back_inserter(keys));
+			break;
+	}
+	lua_pop(L, 1);
+}
+
+// These two are separate functions so I can use a protected call on it to catch errors.
+static int impl_is_deprecated(lua_State* L)
+{
+	auto key = luaL_checkstring(L, 2);
+	auto type = lua_getfield(L, 1, key);
+	if(type == LUA_TTABLE) {
+		lua_pushliteral(L, "__deprecated");
+		if(lua_rawget(L, -2) == LUA_TBOOLEAN) {
+			auto deprecated = luaW_toboolean(L, -1);
+			lua_pushboolean(L, deprecated);
+			return 1;
+		}
+		lua_pop(L, 1);
+	}
+	lua_pushboolean(L, false);
+	return 1;
+}
+
+static int impl_get_dir_suffix(lua_State*L)
```

Comment:
Are these suffixes a convention, either from Lua or another language?

If they're not a convention: for functions, the suffix ` ()` would seem more natural to me than ` ƒ`, although both make sense. I'm not seeing the logic that associates daggers with tables.

General comment on the whole PR: more code-comments would help. Knowing what each function is meant to do makes reviewing both easier and better (because the reviewers don't make as many incorrect assumptions).

Reply:
I chose dagger for table because it resembles a T, and florin for functions because it resembles an F. An italicized lower-case F (which the florin basically is) _is_ commonly used for functions in mathematics, which influenced my decision.

Reply:
Just for the record, I don't actually have any special attachment to these suffixes. If someone has better suggestions, I'm open to changing them.